### PR TITLE
Add Art of Code to Courses

### PR DESCRIPTION
**Pull request by Fazer01**
## What does this PR do?
Add course / point of view on the "Art of Code" by Dylan Beattie
### Description
A brilliant video / course on the Art of Code explained by Dylan Beattie. 

### Why is this valuable (or not)?
Since there was not already a course with this subject I thought it might be valuable to add it. Because of the way of telling and point of view by Dylan, imho will contribute to this great course list.

### How do we know it's really free?
It's on youtube. 

### For book lists, is it a book? For course lists, is it a course? etc.
It is a course. 

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -805,6 +805,7 @@
 ### Misc
 
 * [Advanced Adobe XD (Web Design)](https://webdesign.tutsplus.com/courses/advanced-adobe-xd-for-everyone) - Adi Purdila
+* [Art of Code](https://www.youtube.com/watch?v=6avJHaC3C2U&list=PUTdw38Cw6jcm0atBPA39a0Q) - Dylan Beattie (Youtube)
```
Comment (eshellman) [inline courses/free-courses-en.md:808]: For youtube, we prefer "playlist" links and the name of the producer/creator. docs/CONTRIBUTING.md#platforms-and-access-notes

Comment (Fazer01) [reply courses/free-courses-en.md:808]: Hi @eshellman. The youtube link is a playlist link? And the name of the collaborator / creator is behind the link. Dylan Beattie. 
Do you have an example of how you exactly want it? :) 

Thanks!

Comment (davorpa) [reply courses/free-courses-en.md:808]: Just 1h video. enough for me

Fixing YT url, title and notes...

```suggestion
* [The Art of Code - Dylan Beattie](https://www.youtube.com/watch?v=6avJHaC3C2U) - Dylan Beattie (YouTube)
```

Comment (eshellman) [reply courses/free-courses-en.md:808]: ... and you can omit the (YouTube)

**Comment by Fazer01** [comment]
Provided changes as requested @davorpa

**Review by davorpa — APPROVED** [review]

File: courses/free-courses-en.md (hunk 1 above)
Comment (davorpa) [reply courses/free-courses-en.md]: > ... and you can omit the (YouTube)

Already done by @Fazer01

**Pull request merged**